### PR TITLE
Add EventInvalidSecretRequest

### PR DESCRIPTION
**Pull request by palango**
## Description

During my work on I found out that there is currently no event emitted when a `SecretReveal` is received, which has an amount that is smaller than the intended transfer amount.

This PR adds `EventInvalidSecretRequest` for that case.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [x] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [x] Error conditions are handled
    - [x] Exceptions are propagated to the correct parent greenlet
    - [x] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [x] State changes are forward compatible
    - [x] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [x] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [x] Properly covers the bug
        - [x] If an integration test is used, it could not be written as a unit test
- Documentation
    - [x] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [x] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/transfer/mediated_transfer/initiator.py [hunk 1]
```diff
@@ -370,7 +374,11 @@ def handle_secretrequest(
 
     elif not is_valid_secretrequest and is_message_from_target:
         initiator_state.received_secret_request = True
-        iteration = TransitionResult(initiator_state, list())
+        invalid_request = EventInvalidSecretRequest(
+            intended_amount=initiator_state.transfer_description.amount,
+            actual_amount=state_change.amount,
+        )
+        iteration = TransitionResult(initiator_state, [invalid_request])
```
Comment (pirapira) [inline raiden/transfer/mediated_transfer/initiator.py]: Why is this variable called `iteration`?  Is there something going over something again and again?

Comment (palango) [reply raiden/transfer/mediated_transfer/initiator.py]: I think it relates to the iteration of the state machine?

Comment (pirapira) [reply raiden/transfer/mediated_transfer/initiator.py]: I'll ask further on the knowledge base.

Comment (pirapira) [reply raiden/transfer/mediated_transfer/initiator.py]: Asked https://ethereum.stackexchange.com/questions/74797/in-raidens-source-why-is-a-result-of-state-transition-called-iteration .

File: raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py [hunk 2]
```diff
@@ -1709,7 +1710,7 @@ def test_state_wait_secretrequest_valid_amount_and_fee():
 
     state_change_2 = ReceiveSecretRequest(
         payment_identifier=UNIT_TRANSFER_IDENTIFIER,
-        amount=setup.lock.amount - fee_amount - 1,
+        amount=setup.lock.amount - fee_amount - 1,  # Now the amount becomes to small
```
Comment (pirapira) [inline raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py]: You mean "too small"?

**Comment by pirapira** [comment]
>  Exceptions are propagated to the correct parent greenlet

What does this mean in the checklist?

**Comment by pirapira** [comment]
>  Have good messages

Is a one-line commit message fine?

**Comment by pirapira** [comment]
How is the new event handled?  Fed into the state machine?  For testing that, don't we need an integration test too?

**Comment by palango** [comment]
> Is a one-line commit message fine?

I extended the commit message.

> How is the new event handled? Fed into the state machine? For testing that, don't we need an integration test too?

At the moment this is purely informational. Before a `SecretRequest` with too low amount was just ignored, now this event with the information is logged and will be stored in the database, which should make debugging easier.

**Comment by pirapira** [comment]
I see.  And the state machine silently ignores an unknown event.

**Comment by pirapira** [comment]
When the new event hits `on_raiden_event()` function, https://github.com/raiden-network/raiden/blob/4f5e8fcda420c042eea6dc716d0e02ff6cc636dc/raiden/raiden_event_handler.py#L121
the event will be logged as an unknown event
https://github.com/raiden-network/raiden/blob/4f5e8fcda420c042eea6dc716d0e02ff6cc636dc/raiden/raiden_event_handler.py#L185

**Comment by pirapira** [comment]
So it works as explained I guess.

**Comment by palango** [comment]
> So it works as explained I guess.

Good point, I shoulkd add a simple handler which logs the event.

**Comment by pirapira** [comment]
> I shoulkd add a simple handler which logs the event.

I'm not sure.

**Comment by pirapira** [comment]
This PR is already an improvement and it does what it's supposed to do.

**Comment by pirapira** [comment]
Even better with the handler.

**Pull request merged**